lib/Parse/ParseStmt.cpp

```diff
@@ -1618,12 +1618,23 @@ Parser::parseStmtConditionElement(SmallVectorImpl<StmtConditionElement> &result,
     ThePattern = makeParserResult(AP);
   }
 
-  // Conditional bindings must have an initializer.
+  // Conditional bindings can have the format:
+  //  `let newBinding = <expr>`, or
+  //  `let newBinding`, which is shorthand for `let newBinding = newBinding`
   ParserResult<Expr> Init;
   if (Tok.is(tok::equal)) {
     SyntaxParsingContext InitCtxt(SyntaxContext, SyntaxKind::InitializerClause);
     consumeToken();
     Init = parseExprBasic(diag::expected_expr_conditional_var);
+  } else if (!ThePattern.isNull() && !ThePattern.getPtrOrNull()->getBoundName().empty()) {
+    auto bindingName = new DeclNameRef(ThePattern.getPtrOrNull()->getBoundName());
+    auto loc = new DeclNameLoc(ThePattern.getPtrOrNull()->getEndLoc());
+    auto declRefExpr = new (Context) UnresolvedDeclRefExpr(*bindingName,
+                                                           DeclRefKind::Ordinary,
+                                                           *loc);
```

Comment:
`DeclNameRef` and `DeclNameLoc` should be allocated on the stack.

Reply:
How do you allocate these on the stack instead of on the heap? (Not a C++ expert!)

Also, just curious for my own understanding, why should these be allocated on the stack?

Reply:
> How do you allocate these on the stack instead of on the heap?

As simple as dropping the `new` 🙂. The allocations are not supposed to outlive this local scope anyway, and neither are uniqued or memoized via the `ASTContext`, so there is really no reason to do this on the heap just to dereference and copy them on the next line (otherwise you would have to `delete` them as well!).

Reply:
aha, interesting! thanks